crates/matrix-sdk/src/oidc.rs

```diff
@@ -709,25 +741,75 @@ impl OidcLoginBuilder {
     ///
     /// [`ClientErrorCode`]: matrix_sdk::oidc::types::errors::ClientErrorCode
     #[instrument(target = "matrix_sdk::client", skip_all)]
-    pub async fn login_with_authorization_code<F, Fut>(
+    pub async fn url_for_login_with_authorization_code(
         self,
         redirect_uri: &Url,
-        use_authorization_url: F,
-    ) -> Result<AccessTokenResponse, OidcError>
-    where
-        F: FnOnce(Url, String) -> Fut + Send,
-        Fut: Future<Output = Result<String, OidcError>> + Send,
-    {
+    ) -> Result<OidcAuthenticationUrl, OidcError> {
         let issuer = self.oidc.issuer().await.ok_or(OidcError::MissingAuthenticationIssuer)?;
         info!(issuer, "Logging in via OpenID Connect with the Authorization Code flow");
 
         let scope = generate_scope(self.device_id)?;
 
-        let response = self
-            .oidc
-            .authorize_scope_with_authorization_code(&scope, redirect_uri, use_authorization_url)
-            .await?;
+        self.oidc.url_for_authorize_scope_with_authorization_code(&scope, redirect_uri).await
+    }
 
+    /// Login via the Authorization Code flow.
+    ///
+    /// # Arguments
+    ///
+    /// * `code` - The code received as part of the redirect URI when login was
+    ///   successful.
+    ///
+    /// * `state` - The unique string that was included in the
+    ///   `OidcAuthenicationUrl` used for login.
+    ///
+    /// This method should be called after presenting the URL returned by
```

Comment:
Can you put that above the arguments? Also same comment, it should be called after the user being redirected.